src/sockets/socket.cpp

```diff
@@ -12,6 +12,16 @@
 #include <string>
 #include <utility>
 
+
+#ifndef _WIN32
+#ifdef SOCKETS_USE_MUSL
+#define IPV6_INFO_PORT(ipv6_info) ((ipv6_info)->sin6_addr.__in6_union.__s6_addr)
```

Comment:
this is not needed. ipv6_info->sin6_addr.s6_addr is a portable name which works for both glibc and musl. but yeah, getting rid of the implementation details from the libc is probably a good idea

Reply:
Oh, I'll look into that

Reply:
You are indeed right, thanks for the help 😄